issue/1393-v4-stats-timezone-fix

anitaa1990 opened the pull request:
This PR Fixes #1393. The fix is purely a FluxC side fix and this PR just updates the FluxC hash to the app.

~~This PR is in draft till this [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1375) is merged.~~ More details on the issue can be found [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1375).

### Testing
It would be good to test with multiple timezones to verify if the dates passed to the API request is correct. I changed the timezone of my testing device to GMT+14 and verify that the date bar displays the correct start date for the current day, this week, this month and this year.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

anitaa1990 committed:
Updated fluxC hash for timezone fix

anitaa1990 committed:
Updated the FluxC hash to include the timezone fix for stats v4 UI

0nko commented:
I can still see the bug. I first ran the app with automatic timezone (set to Czech Republic) and the 1st day was Sep 1. Then I changed the timezone to Macau and it was Aug 31:

<img src="https://user-images.githubusercontent.com/1522856/64195666-55ad4580-ce82-11e9-8744-da4326efa00c.png" width="320" />

anitaa1990 committed:
Added logic to calculate bar label count based on the chart size to avoid duplicate x-axis labels

anitaa1990 committed:
Fixed timezone issue bug when formatting display date for week and month tab

anitaa1990 commented:
Thanks for testing @Onko. Debugging this further in Stethos, I noticed that the date being passed to the API request is the correct date i.e. 2019-09-01 -> (this was the bug that is getting fixed by this PR. Previously, the date was being passed as 2019-08-31). 

That being said, I wasn't able to reproduce the bug when switching from my own timezone (GMT+5.30). I did reproduce it though by following the steps below: 
- Switch the timezone to GMT+2 and open the app.
- Switch timezone to GMT+14 and open the app again.
- Notice the date bar and the x-axis labels displayed the previous month last date. This was happening for `This week` and `This month` tab.

The bug was because the date being formatted for display in the date bar and the x-axis labels were formatting it to the previous day instead of the actual date. For instance, even though the date passed to this method was 2019-09-01, the formatted date returned the value Aug 31 instead. I fixed that in 74ee0e1.

I also noticed from the screenshot above that the x-axis labels were displaying the same labels twice and fixed it in dd4b56c.

Nice catch! PR ready for another round 👍

0nko reviewed (approved):
Thanks for the changes, Anitaa! Looks good now 👍
